src/Native/System.Native/pal_networking.cpp

```diff
@@ -1188,7 +1188,11 @@ extern "C" Error SystemNative_GetIPv4MulticastOption(int32_t socket, int32_t mul
         return PAL_EINVAL;
     }
 
+#if HAVE_IP_MREQ
+    ip_mreq opt;
+#elif HAVE_IP_MREQN
```

Comment:
I'd prefer it if in general we make the first check be the thing that goes to the code that's already there, e.g. have this be:
```C++
#if HAVE_IP_MREQN
    ip_mreqn opt;
#else
    ip_mreq opt;
#endif
```
This will help to minimize the chances that we accidentally break one of our officially supported platforms while bringing up support for additional platforms.  As necessary it can always be refactored later.